From 23469dfa83a2a09fb0503122b59767c3b3d60125 Mon Sep 17 00:00:00 2001
From: Marie Ramlow <me@nycode.dev>
Date: Sun, 8 Jun 2025 00:18:16 +0200
Subject: [PATCH] fix: sort contracts for deterministic build

---
 packaging/buildTasks.mjs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packaging/buildTasks.mjs b/packaging/buildTasks.mjs
index 4a30bbf..6aaa709 100644
--- a/packaging/buildTasks.mjs
+++ b/packaging/buildTasks.mjs
@@ -124,6 +124,7 @@ export async function packResources() {
     const start = Date.now()
 
     const contracts = await glob("contractdata/**/*.json")
+    contracts.sort();
     const b = []
     const el = []
 
-- 
2.49.0

